library/Core.cpp

```diff
@@ -947,8 +947,8 @@ command_result Core::runCommand(color_ostream &con, const std::string &first_, v
                 << "Supported keys: [Ctrl-][Alt-][Shift-](A-Z, 0-9, F1-F12, `, or Enter)." << endl
                 << "Context may be used to limit the scope of the binding, by" << endl
                 << "requiring the current context to have a certain prefix." << endl
-                << "Current UI context is: "
-                << Gui::getFocusString(Core::getTopViewscreen()) << endl;
+                << "Current UI context is: " << endl
+                << join_strings("\n", Gui::getFocusString(Gui::getDFViewscreen())) << endl;
```

Comment:
I think this should be `join_strings("\n", Gui::getFocusString(Gui::getTopViewscreen(true)))`. That is:
1. ignore dismissed viewscreens at the top of the stack
2. unfocused ZScreens are ignored when determining the focus string (this might take changes to the Lua ZScreen implementation)
3. the viewscreen used to determine the focus string is the focused ZScreen (if any) or the first non-ZScreen on the stack. if that screen is a DF viewscreen, then further context can be determined

This will make the focus string correspond to the viewscreen that actually has keyboard focus

this might take some explaining, but I wanted to get the requirements written down here first

Reply:
Do you want me to look into the ZScreen stuff you mentioned here or is that the stuff you wanna work on after the focus string stuff is merged?

Reply:
I can work on that after this PR is merged. That functionality is tied to how hotkeys will work when ZScreens of different properties are onscreen.